### PR TITLE
rust - dns - v9

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: #2686 

Changes from last PR:
- Add -lm on Linux
- Create a LoggerFlags type to encapsulate tracking loggers
- Type filtering for DNS logs (I thought it was working before, but only request/reply filtering was).

TODOs:
- Do a lib check in configure for extra libraries added.
- Not all DNS types are logged (all comon ones are).
- Add command that generated files to generated files (forgot).
- Post parse validation so we can log and alert on things like an invalid SSHFP (this would be a new feature, as this isn't done in the C version either).
- Look further into Rust documentation generation. While I had it somewhat work, no index page was created.

Prscript:
- PR jasonish-pcap: https://buildbot.openinfosecfoundation.org/builders/jasonish-pcap/builds/148
- PR jasonish: https://buildbot.openinfosecfoundation.org/builders/jasonish/builds/500

### CODE REVIEW

File: rust/src/dns/log.rs [hunk 1]
```diff
@@ -22,7 +22,252 @@ use std::string::String;
...
+pub const LOG_CDS        : u64 = BIT_U64!(52);
+pub const LOG_CDNSKEY    : u64 = BIT_U64!(53);
+pub const LOG_SPF        : u64 = BIT_U64!(54);
+pub const LOG_TKEY       : u64 = BIT_U64!(55);
+pub const LOG_TSIG       : u64 = BIT_U64!(56);
+pub const LOG_MAILA      : u64 = BIT_U64!(57);
+pub const LOG_ANY        : u64 = BIT_U64!(58);
+pub const LOG_URI        : u64 = BIT_U64!(59);
+
+fn dns_log_rrtype_enabled(rtype: u16, flags: u64) -> bool
```
Comment (inliniac) [inline rust/src/dns/log.rs]: isn't this going to be slow?

Comment (jasonish) [inline rust/src/dns/log.rs]: It's just a direct port of what's in C now for 1:1 behaviour.

**Comment by inliniac** [comment]
On 10-05-17 18:01, Jason Ish wrote:
> *@jasonish* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In rust/src/dns/log.rs
> <https://github.com/inliniac/suricata/pull/2703#discussion_r115781965>:
> 
>> +pub const LOG_DNSKEY     : u64 = BIT_U64!(46);
> +pub const LOG_DHCID      : u64 = BIT_U64!(47);
> +pub const LOG_NSEC3      : u64 = BIT_U64!(48);
> +pub const LOG_NSEC3PARAM : u64 = BIT_U64!(49);
> +pub const LOG_TLSA       : u64 = BIT_U64!(50);
> +pub const LOG_HIP        : u64 = BIT_U64!(51);
> +pub const LOG_CDS        : u64 = BIT_U64!(52);
> +pub const LOG_CDNSKEY    : u64 = BIT_U64!(53);
> +pub const LOG_SPF        : u64 = BIT_U64!(54);
> +pub const LOG_TKEY       : u64 = BIT_U64!(55);
> +pub const LOG_TSIG       : u64 = BIT_U64!(56);
> +pub const LOG_MAILA      : u64 = BIT_U64!(57);
> +pub const LOG_ANY        : u64 = BIT_U64!(58);
> +pub const LOG_URI        : u64 = BIT_U64!(59);
> +
> +fn dns_log_rrtype_enabled(rtype: u16, flags: u64) -> bool
> 
> It's just a direct port of what's in C now for 1:1 behaviour.

Yikes, you're right. I thought it was just like this at start up. Oh well.

**Comment by jasonish** [comment]
It is something I want to revisit/rethink when there is only a rust implementation to worry about. Maybe sooner tho.

**Comment by inliniac** [comment]
On 10-05-17 18:14, Jason Ish wrote:
> It is something I want to revisit/rethink when there is only a rust
> implementation to worry about. Maybe sooner tho.

Lets make a note and first continue with the Rust integration steps.
We'll optimize later.

**Comment by jasonish** [comment]
See: https://github.com/inliniac/suricata/pull/2706

**Pull request closed without merging**